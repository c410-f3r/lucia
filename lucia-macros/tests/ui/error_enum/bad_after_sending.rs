#[lucia_macros::pkg(api(Foo), data_format(json))]
mod pkg {
  #[pkg::after_sending]
  async fn after_sending(foo: i32) -> lucia::Result<()> {
    Ok(())
  }

  #[pkg::req_data]
  struct ReqData;

  #[pkg::res_data]
  struct ResData;
}

fn main() {
}
