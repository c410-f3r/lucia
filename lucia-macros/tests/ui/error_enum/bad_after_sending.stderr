error: `#[pkg::after_sending]` must be an async function named `after_sending` containing any combination of `api: &mut SomeApi`, `params: &mut SomePackageParams`, `req_bytes: &[u8]`, and `req_params: &mut SomeRequestParams`.
 --> tests/ui/error_enum/bad_after_sending.rs:6:10
  |
6 |   #[pkg::after_sending]
  |          ^^^^^^^^^^^^^
